train/train.py

```diff
@@ -284,13 +308,50 @@ def train():
 
     random.shuffle(raw_data)
 
-    dataset = CustomDataset(raw_data, tokenizer)
+    if training_args.do_eval:
+        # Take 90:10 train-validation split
+        raw_train_data, raw_eval_data = split_data(raw_data, data_args.data_path, 0.9)
+        train_dataset = CustomDataset(raw_train_data, tokenizer)
+        eval_dataset = CustomDataset(raw_eval_data, tokenizer)
+    else:
+        train_dataset = CustomDataset(raw_data, tokenizer)
+        
+    def preprocess_logits_for_metrics(logits, labels):
+        pred_ids = torch.argmax(logits, dim=-1)
+        shift_logits = logits[..., :-1, :].contiguous()
+        shift_labels = labels[..., 1:].contiguous()
+        loss_fct = CrossEntropyLoss(reduction="none")
```

Comment:
If you mean loss function, can we rename this to "loss_fn" or "loss_function" to keep consistency on the names.